docs/customruns.md

```diff
@@ -0,0 +1,413 @@
+`CustomRun`s which reference their type. If no such controller is running, `CustomRun`s
+will have no `.status` value and no further action will be taken.
+
+## Configuring a `CustomRun`
+
+A `CustomRun` definition supports the following fields:
+
+- Required:
+  - [`apiVersion`][kubernetes-overview] - Specifies the API version, for example
+    `tekton.dev/v1beta1`.
+  - [`kind`][kubernetes-overview] - Identifies this resource object as a `CustomRun` object.
+  - [`metadata`][kubernetes-overview] - Specifies the metadata that uniquely identifies the
+    `CustomRun`, such as a `name`.
+  - [`spec`][kubernetes-overview] - Specifies the configuration for the `CustomRun`.
+    - [`customRef`](#1-specifying-the-target-custom-task-with-customref) - Specifies the type and
+      (optionally) name of the custom task type to execute.
+    - [`customSpec`](#2-specifying-the-target-custom-task-by-embedding-its-spec) - Embed the custom task resource spec
+      directly in a `CustomRun`.
+- Optional:
+  - [`timeout`](#specifying-timeout) - specifies the maximum duration of a single execution
+    of a `CustomRun`.
+  - [`retries`](#specifying-retries) - specifies the number of retures upon `CustomRun`failure.
+  - [`params`](#specifying-parameters) - Specifies the desired execution
+    parameters for the custom task.
+  - [`serviceAccountName`](#specifying-a-serviceaccount) - Specifies a `ServiceAccount`
+    object that provides custom credentials for executing the `CustomRun`.
+  - [`workspaces`](#specifying-workspaces) - Specifies the physical volumes to use for the
+    [`Workspaces`](workspaces.md) required by a custom task.
+
+[kubernetes-overview]:
+  https://kubernetes.io/docs/concepts/overview/working-with-objects/kubernetes-objects/#required-fields
+
+### Specifying the target Custom Task
+
+A custom task resource's `CustomSpec` may be directly embedded in the `CustomRun` or it may
+be referred to by a `CustomRef`. But, not both at the same time.
+
+1. [Specifying the target Custom Task with customRef](#1-specifying-the-target-custom-task-with-customref)
+   Referring a custom task (i.e. `CustomRef` ) promotes reuse of custom task definitions.
+
+2. [Specifying the target Custom Task by embedding its spec](#2-specifying-the-target-custom-task-by-embedding-its-spec)
+   Embedding a custom task (i.e. `CustomSpec` ) helps in avoiding name collisions with other users within the same namespace.
+   Additionally, in a pipeline with multiple embedded custom tasks, the details of entire pipeline can be fetched in a
+   single API request.
+
+#### Specifying the target Custom Task with customRef
+
+To specify the custom task type you want to execute in your `CustomRun`, use the
+`customRef` field as shown below:
+
+```yaml
+spec:
+  customRef:
+    apiVersion: example.dev/v1alpha1
+    kind: Example
+```
+
+This initiates the execution of a `CustomRun` of a custom task of type `Example`, in
```

Comment:
Suggest clarifying the wording here, as this sounds a bit like Tekton will initiate the execution of a customrun when it is created, rather than the custom task controller being responsible for executing it.

Reply:
SG, rephrased it~